components/wasm-opt-cxx-sys/src/shims.h

```diff
@@ -43,6 +44,98 @@ namespace wasm_shims {
   }
 }
 
+namespace wasm_shims {
+  struct InliningOptions {
+    wasm::InliningOptions inner;
+
+    void setAlwaysInlineMaxSize(uint32_t size) {
+      inner.alwaysInlineMaxSize = size;
+    }
+
+    void setOneCallerInlineMaxSize(uint32_t size) {
+      inner.oneCallerInlineMaxSize = size;
+    }
+
+    void setFlexibleInlineMaxSize(uint32_t size) {
+      inner.flexibleInlineMaxSize = size;
+    }
+
+    void setAllowFunctionsWithLoops(bool allow) {
+      inner.allowFunctionsWithLoops = allow;
+    }
+
+    void setPartialInliningIfs(uint32_t number) {
+      inner.partialInliningIfs = number;
+    }
+  };
+    
+  std::unique_ptr<InliningOptions> newInliningOptions() {
+    return std::make_unique<InliningOptions>();
+  }
+}
+
+namespace wasm_shims {
+  struct PassOptions {
+    wasm::PassOptions inner;
+
+    void setValidate(bool validate) {
+      inner.validate = validate;
+    }
+
+    void setValidateGlobally(bool validate) {
+      inner.validateGlobally = validate;
+    }
+
+    void setOptimizeLevel(int32_t level) {
+      inner.optimizeLevel = level;
+    }
+
+    void setShrinkLevel(int32_t level) {
+      inner.shrinkLevel = level;
+    }
+
+    void setInliningOptions(std::unique_ptr<wasm_shims::InliningOptions> inlining) {
+      inner.inlining = inlining->inner;
+    }
+    
+    void setTrapsNeverHappen(bool ignoreTraps) {
+      inner.trapsNeverHappen = ignoreTraps;
+    }
+
+    void setLowMemoryUnused(bool memoryUnused) {
+      inner.lowMemoryUnused = memoryUnused;
+    }
+
+    void setFastMath(bool fastMath) {
+      inner.fastMath = fastMath;
+    }
+
+    void setZeroFilledMemory(bool zeroFilledMemory) {
+      inner.zeroFilledMemory = zeroFilledMemory;
+    }
+
+    void setDebugInfo(bool debugInfo) {
+      inner.debugInfo = debugInfo;
+    }
+
+    void setGenerateStackIR(bool generateStackIR) {
+      inner.generateStackIR = generateStackIR;
+    }
+
+    void setOptimizeStackIR(bool optimizeStackIR) {
+      inner.optimizeStackIR = optimizeStackIR;
+    }
+
+    void setArguments(std::string& key, std::string& value) {
+      inner.arguments[std::move(key)] = std::move(value);
+    }
+  };
+
+  std::unique_ptr<PassOptions> newPassOptions() {
+    return std::make_unique<PassOptions>();
+  }
+}
```

Comment:
Had to move these 2 structs up since `ModuleWriter` requires uses `PassOptions` now.